Add e2b.app

sitole opened the pull request:
# Public Suffix List (PSL) Submission

### Checklist of required steps

* [X] Description of Organization
* [X] Robust Reason for PSL Inclusion
* [X] DNS verification via dig
* [X] Each domain listed in the PRIVATE section has and shall maintain at least two years remaining on registration, and we shall keep the `_psl` TXT record in place in the respective zone(s).

__Submitter affirms the following:__ 

 * [X] We are listing *any* third-party limits that we seek to work around in our rationale such as those between IOS 14.5+ 
 * [X] This request was _not_ submitted with the objective of working around other third-party limits.
 * [X] The submitter acknowledges that it is their responsibility to maintain the domains within their section. This includes removing names which are no longer used, retaining the _psl DNS entry, and responding to e-mails to the supplied address. Failure to maintain entries may result in removal of individual entries or the entire section.
 * [X] The [Guidelines](https://github.com/publicsuffix/list/wiki/Guidelines) were carefully _read_ and _understood_, and this request conforms to them.
 * [X] The submission follows the [guidelines](https://github.com/publicsuffix/list/wiki/Format) on formatting and sorting.
 * [X] A role-based email address has been used and this inbox is actively monitored with a response time of no more than 30 days.

**Abuse Contact:**

* [X] Abuse contact information (email or web form) is available and easily accessible.

We have mail dedicated for reporting security / abusing at `security@e2b.dev`.

---

For PRIVATE section requests that are submitting entries for domains that match their organization website's primary domain, please understand that this can have impacts that may not match the desired outcome and take a long time to rollback, if at all.

To ensure that requested changes are entirely intentional, make sure that you read the affectation and propagation expectations, that you understand them, and confirm this understanding. 

PR Rollbacks have lower priority, and the volunteers are unable to control when or if browsers or other parties using the PSL will refresh or update.

(Link: [about propagation/expectations](https://github.com/publicsuffix/list/wiki/Guidelines#appropriate-expectations-on-derivative-propagation-use-or-inclusion))

 * [X] *Yes, I understand*. I could break my organization's website cookies and cause other issues, and the rollback timing is acceptable. *Proceed anyways*.
---

## Description of Organization

I'm on the infra team at E2B. It allows you to create virtual machines on request and supports code evaluation, running publicly facing services, etc. Each machine is assigned dedicated subdomains, which can contain customers data. Therefore, it's important to ensure that customers cannot interact (e.g., share cookies) between these subdomains.

**Organization Website:**
https://e2b.dev/

## Reason for PSL Inclusion
We want our users subdomains to be isolated for other customers (cookies, password managers, etc).

**Number of users this request is being made to serve:**
Currently we are serving millions of short-lived machines with their own subdomain every week.

## DNS Verification
```
dig +short TXT _psl.e2b.app
"https://github.com/publicsuffix/list/pull/2431"
```

fakeboboliu commented:
Here's a checklist:

* [ ]  Expiration (Note: Must STAY >2y at all times)
  * [ ]  e2b.app`Registry Expiry Date: 2026-01-20T01:18:57Z`
* [x]  DNS _psl entries (Note: Must STAY in place)
* [ ]  Reasoning/Organization description
  - No subdomains can be found through subdomainfinder, which could be reasonable since short-lived VMs can shut down before any clue leaks to the internet.
  - No CT found, that's a red flag, `.app` are wholly HSTS-prelisted - thus it should not work without a certificate.
* [x]  Non-personal email address
* [x]  Abuse contact
  e2b.app redirects to e2b.dev, where `hello@e2b.dev` can be found on the footer.

---

Please use `e2b.app` instead of `*.e2b.app` if your user's subdomain would be `somethingunique.e2b.app`

sitole commented:
> Expiration (Note: Must STAY >2y at all times)

Fixed, thanks. 

> Please use `e2b.app` instead of `*.e2b.app` if your user's subdomain would be `somethingunique.e2b.app`

Fixed

> No CT found for suffix *.e2b.app

Can you please explain it more? I found CTs to be present (https://crt.sh/?q=%25.e2b.app). Currently wea reusing Google Cloud and Cloudflare for certificates that should be automatically propagated here.

cc @fakeboboliu

fakeboboliu commented:
> > No CT found for suffix *.e2b.app
> 
> Can you please explain it more? I found CTs to be present (https://crt.sh/?q=%25.e2b.app). Currently wea reusing Google Cloud and Cloudflare for certificates that should be automatically propagated here.

It's for entry `*.e2b.app`. You can ignore it since you've changed your listed suffix from `*.e2b.app` to `e2b.app`.

But the new issue is: CT shows only one wildcard cert. Does that mean you're in control of the codes behind all the short-lived subdomains?

If so, you'd better try `__Host-` cookies rather than PSL, which needs a long time to be propagated through your client's browsers - which can be extremely long, especially for enterprise users who rarely upgrade their browsers.

sitole commented:
> > > No CT found for suffix *.e2b.app
> > 
> > 
> > Can you please explain it more? I found CTs to be present (https://crt.sh/?q=%25.e2b.app). Currently wea reusing Google Cloud and Cloudflare for certificates that should be automatically propagated here.
> 
> It's for entry `*.e2b.app`. You can ignore it since you've changed your listed suffix from `*.e2b.app` to `e2b.app`.
> 
> But the new issue is: CT shows only one (well, two, but you know what I mean) wildcard cert. Does that mean you're in control of the codes behind all the short-lived subdomains?
> 
> If so, you'd better try `__Host-` cookies rather than PSL, which needs a long time to be propagated through your client's browsers - which can be extremely long, especially for enterprise users who rarely upgrade their browsers.

Maybe i was not clear enough with description of what we are doing with subdomain. Our clients will receive VM and they can build their things inside but only incoming traffic that we are allowing is via HTTP protocol. Basically when you will request VM and ask for subdomain for port 8080 you will received something like `8080-<sandboxId>.e2b.app`. We are providing http based traffic proxy for requests thus we have just one wildcard certificate for all subdomains.

I think we need `*.e2b.app` back then if i understand correctly.

fakeboboliu commented:
> Maybe i was not clear enough with description of what we are doing with subdomain. Our clients will receive VM and they can build their things inside but only incoming traffic that we are allowing is via HTTP protocol. Basically when you will request VM and ask for subdomain for port 8080 you will received something like 8080-<sandboxId>.e2b.app. We are providing http based traffic proxy for requests thus we have just one wildcard certificate for all subdomains.

You may still try adding `__Host-` prefix to cookies at your HTTP gateway. It will provide an immediate effect(and is more flexible than PSL).
This will not be one of the key criteria for PSL inclusion review anyway, so calm down.

> I think we need *.e2b.app back then if i understand correctly.

No. I'll repeat it: please use `e2b.app` instead of `*.e2b.app` if your user's subdomain would be `somethingunique.e2b.app`.

sitole commented:
> > Maybe i was not clear enough with description of what we are doing with subdomain. Our clients will receive VM and they can build their things inside but only incoming traffic that we are allowing is via HTTP protocol. Basically when you will request VM and ask for subdomain for port 8080 you will received something like 8080-.e2b.app. We are providing http based traffic proxy for requests thus we have just one wildcard certificate for all subdomains.
> 
> You may still try adding `__Host-` prefix to cookies at your HTTP gateway. It will provide an immediate effect(and is more flexible than PSL). This will not be one of the key criteria for PSL inclusion review anyway.

Can try to discuss it in team but still not covering all things that listing domain brings to us so should continue for now.

> > I think we need *.e2b.app back then if i understand correctly.
> 
> No. I'll repeat it: please use `e2b.app` instead of `*.e2b.app` if your user's subdomain would be `somethingunique.e2b.app`.

Done, thx

simon-friedberger commented:
Can somebody explain to me how .app requires HTTPS but only HTTP is offered?

sitole commented:
> Can somebody explain to me how .app requires HTTPS but only HTTP is offered?

Do you mean our use-case or talking about something different? We are offering HTTPS-only traffic.

> Fix sorting tag

Can you please explain what exact sorting here is wrong?
We have only one entry so there is no sorting, in list its sorted by company name right after `Forgerock` company with same prefix `Fo`. Is there some script that i can run for checkin correct sorting / file formatting? Thanks.

cc @simon-friedberger
